src/components/view/carousel.tsx

```diff
@@ -0,0 +1,80 @@
+"use client";
+import React, { useEffect } from "react";
+
+import LeftIcon from "@/assets/icons/left.svg";
+import RightIcon from "@/assets/icons/right.svg";
+
+interface CarouselProps {
+  items: React.ReactNode[];
+  initialScroll?: number;
+}
+
+export const Carousel = ({ items, initialScroll = 0 }: CarouselProps) => {
+  const carouselRef = React.useRef<HTMLDivElement>(null);
+  const [canScrollLeft, setCanScrollLeft] = React.useState(false);
+  const [canScrollRight, setCanScrollRight] = React.useState(true);
+
+  useEffect(() => {
+    if (carouselRef.current) {
+      carouselRef.current.scrollLeft = initialScroll;
+      checkScrollability();
+    }
+  }, [initialScroll]);
+
+  const checkScrollability = () => {
+    if (carouselRef.current) {
+      const { scrollLeft, scrollWidth, clientWidth } = carouselRef.current;
+      setCanScrollLeft(scrollLeft > 0);
+      setCanScrollRight(scrollLeft < scrollWidth - clientWidth);
+    }
+  };
```

Comment:
ここのメモ化もお願いします！（注: `useCallback`のインポートが必要です）

> [!TIP]
>
> 順番を入れ替えたのは以下エラーが出ちゃうためです！
>
> ```console
> Block-scoped variable 'checkScrollability' used before its > declaration.
> ```

```suggestion
  const checkScrollability = useCallback(() => {
    if (carouselRef.current) {
      const { scrollLeft, scrollWidth, clientWidth } = carouselRef.current;
      setCanScrollLeft(scrollLeft > 0);
      setCanScrollRight(scrollLeft < scrollWidth - clientWidth);
    }
  }, []);

  useEffect(() => {
    if (carouselRef.current) {
      carouselRef.current.scrollLeft = initialScroll;
      checkScrollability();
    }
  }, [checkScrollability, initialScroll]);
```